crates/core/src/factories/mem_transport.rs

```diff
@@ -154,23 +166,52 @@ impl TxImp for MemTransport {
     }
 }
 
-type Res = tokio::sync::oneshot::Sender<K2Result<()>>;
+/// A oneshot sender to report back a K2Result.
+type ResultSender = tokio::sync::oneshot::Sender<K2Result<()>>;
+
+/// The sending end of the channel to send commands to a MemTransport's
+/// command runner task.
 type CmdSend = tokio::sync::mpsc::UnboundedSender<Cmd>;
+
+/// The receiving end of the channel to send commands to a MemTransport's
+/// command runner task.
 type CmdRecv = tokio::sync::mpsc::UnboundedReceiver<Cmd>;
-type DataSend = tokio::sync::mpsc::UnboundedSender<(bytes::Bytes, Res)>;
-type DataRecv = tokio::sync::mpsc::UnboundedReceiver<(bytes::Bytes, Res)>;
-type ConSend = tokio::sync::mpsc::UnboundedSender<(Url, DataSend, DataRecv)>;
-type ConRecv = tokio::sync::mpsc::UnboundedReceiver<(Url, DataSend, DataRecv)>;
 
-struct DropSend {
-    send: DataSend,
+/// The sending end of a "data" channel between two MemTransport instances.
+/// Used to send the bytes of a message alongside with a ResultSender to
+/// report back the result of handling the message.
+type DataSend =
+    tokio::sync::mpsc::UnboundedSender<(bytes::Bytes, ResultSender)>;
+
+/// The receiving end of a "data" channel between two MemTransport instances.
+/// Used to send the bytes of a message alongside with a ResultSender to
+/// report back the result of handling the message.
+type DataRecv =
+    tokio::sync::mpsc::UnboundedReceiver<(bytes::Bytes, ResultSender)>;
+
+/// The sending end of a "connection establishment channel" used to establish
+/// connections (in the form of two oppsite "data" channels) between different
+/// MemTransport instances.
+type ConnectionSend =
+    tokio::sync::mpsc::UnboundedSender<(Url, DataSend, DataRecv)>;
+
+/// The receiving end of a "connection establishment channel" used to establish
+/// connections (in the form of two oppsite "data" channels) between different
+/// MemTransport instances.
+type ConnectionRecv =
+    tokio::sync::mpsc::UnboundedReceiver<(Url, DataSend, DataRecv)>;
+
+/// An open connection to another peer, containing the sending end of the
+/// "data" channel connected to their MemTransport instance.
+struct OpenConection {
```

Comment:
typo connection

Reply:
The second comment from @ThetaSinner only showed up in the github UI after I added above message. I agree with renaming to `DropConnection`, that makes sense. I updated that in [this fixup commit](https://github.com/holochain/kitsune2/pull/323/commits/228ccc2f4fd4422cf6a2e4197438e127ba3edb81)